src/key_value_store.js

```diff
@@ -206,26 +209,19 @@ export class KeyValueStoreLocal {
         validateGetValueParams(key);
 
         return this.initializationPromise
-            .then(() => {
-                const filePath = path.resolve(this.localEmulationPath, key);
-                const promises = LOCAL_FILE_TYPES.map(({ extension }) => {
-                    return readFilePromised(`${filePath}.${extension}`).catch(() => null);
-                });
-
-                return Promise.all(promises);
-            })
+            .then(() => readdirPromised(this.localEmulationPath))
             .then((files) => {
-                let body = null;
-
-                LOCAL_FILE_TYPES.some(({ contentType }, index) => {
-                    if (files[index] !== null) {
-                        body = parseBody(files[index], contentType);
-
-                        return true;
-                    }
-                });
-
-                return body;
+                const regex = getFileNameRegexp(key);
```

Comment:
This is crazily inefficient and will choke on directories with many files. Perhaps we can keep the previous quick lookup for the most common file extensions (json, png, jpg, html, ...) and fallback to the slow one if nothing is found?